File: src/main/java/com/example/searchingevents/models/dto/EventDTO.java
package com.example.searchingevents.models.dto;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class EventDTO {
    private Long id;
    private Long chatId;
    private Integer messageId;
    private String rawText;
    private String eventType;
    private LocalDateTime eventDateTime;
    private List<TicketOptionDTO> ticketOptions;
    private String city;
    private String url;
    private LocalDateTime createdAt;
}

File: event-client/src/app/search/model/EventDTO.ts
import {TicketOptionDTO} from './TicketOptionDTO';

export interface EventDTO {
  id: number;
  chatId: number | null;
  messageId: number | null;
  rawText: string | null;

  eventType: string;
  eventDateTime: string;
  city: string;
  url: string;
  createdAt: string;

  ticketOptions: TicketOptionDTO[];
}
